epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/launch_plugin.py

```diff
@@ -32,22 +32,17 @@ def compile(self, *args, **opts):
             self.options['nb_core'] = multiprocessing.cpu_count()    
         if 'cwd' in opts and os.path.basename(opts['cwd']) == 'Source':
             path = pjoin(opts['cwd'], 'make_opts')
-            avx_level = self.run_card['avx_level'] if self.run_card['avx_level'] != 'auto' else ''
             common_run_interface.CommonRunCmd.update_make_opts_full(path,
-                {'FPTYPE': self.run_card['floating_type'],
-                 'AVX': avx_level })
+                {'FPTYPE': self.run_card['floating_type'] })
             misc.sprint('FPTYPE checked')
+        cudacpp_supported_backends = [ 'fortran', 'cuda', 'hip', 'cpp', 'cppnone', 'cppsse4', 'cppavx2', 'cpp512y', 'cpp512z', 'cppauto' ]
         if args and args[0][0] == 'madevent' and hasattr(self, 'run_card'):            
-            cudacpp_backend = self.run_card['cudacpp_backend'].upper() # the default value is defined in banner.py
+            cudacpp_backend = self.run_card['cudacpp_backend'].lower() # the default value is defined in launch_plugin.py
             logger.info("Building madevent in madevent_interface.py with '%s' matrix elements"%cudacpp_backend)
-            if cudacpp_backend == 'FORTRAN':
-                args[0][0] = 'madevent_fortran_link'
-            elif cudacpp_backend == 'CPP':
-                args[0][0] = 'madevent_cpp_link'
-            elif cudacpp_backend == 'CUDA':
-                args[0][0] = 'madevent_cuda_link'
+            if cudacpp_backend in cudacpp_supported_backends :
+                args[0][0] = 'madevent_' + cudacpp_backend + '_link'
             else:
-                raise Exception("Invalid cudacpp_backend='%s': only 'FORTRAN', 'CPP', 'CUDA' are supported")
+                raise Exception( "Invalid cudacpp_backend='%s': supported backends are %s"%supported_backends )
             return misc.compile(nb_core=self.options['nb_core'], *args, **opts)
         else:
             return misc.compile(nb_core=self.options['nb_core'], *args, **opts)
```

Comment:
Is this not identical to the change in madevent_interface? Is those line not enough?

Reply:
Hi @oliviermattelaer thanks. Also for this I have added many comments about this in a new issue https://github.com/madgraph5/madgraph4gpu/issues/844.